yt/visualization/accumulators.py

```diff
@@ -0,0 +1,290 @@
+        use_log = [False for i in range(len(field))]
+        field = [self.ad._determine_fields(f)[0] for f in field]
+        tree.set_fields(field, use_log, False)
+        return tree
+
+    def _get_path_field_values(self, tree, path, idx, vals, npts):
+        r"""
+        Determines the value of the field at each point along the path.
+
+        Parameters
+        ----------
+        tree : AMRKDTree
+            An AMRKDTree for the dataset containing the given field values.
+
+        path : np.ndarray
+            The N x 3 array containing the x, y, and z coordinates of the
+            N points used to define the path.
+
+        Returns
+        -------
+        path_field_values : np.ndarray
+            The array containing the field values at each point along the
+            path.
+        """
+        node = tree.locate_node(path[idx])
+        # Data is a list, with one element for each field component
+        data = node.data.my_data
+        # Number of cells in the node
+        dims = data[0].shape
+        # Put the data in a more convenient form: dims x ndims array
+        data = [d.flatten() for d in data]
+        data = ustack(data, axis=1)
+        # Cell width
+        node_left_edge = self.ds.arr(node.get_left_edge(), "code_length")
+        node_right_edge = self.ds.arr(node.get_right_edge(), "code_length")
+        cell_size = (node_right_edge - node_left_edge) / dims
+        while idx < npts:
+            # Make sure point is within domain
+            left_check = path[idx] < self.left_edge
+            right_check = path[idx] >= self.right_edge
+            if np.any(left_check | right_check):
+                msg = f"Point `{path[idx]}` at index `{idx}` outside domain bounds."
+                msg += f"LE: `{self.left_edge}`, RE: `{self.right_edge}`"
+                raise ValueError(msg)
+            # Figure out which cell in the node the point falls within
+            # Origin of node can be offset from origin of origin of volume,
+            # so we have to subtract it off to get the right cell indices
+            cell_ind = ((path[idx] - node_left_edge) / cell_size).astype('i8')
+            # Access the value of the field at that index. Accessing a single
+            # element of a multi-dimensional array using another array is
+            # problematic. Flatten and use a row-major index, indstead
+            I = np.ravel_multi_index(cell_ind, dims)
+            # Get the value of each component for the current cell
+            vals[idx] = data[I]
+            # See if next point is still within the same node (if the next point
+            # is still in range of the array)
+            idx += 1
+            if idx != npts:
+                left_check = path[idx] < node_left_edge
+                right_check = path[idx] >= node_right_edge
+                if np.any(left_check | right_check):
+                    vals, idx = self._get_path_field_values(tree, path, idx, vals, npts)
```

Comment:
It seems to me that this might also be doable outside of the loop.  So do a check if it's inside, and if it's not, call the function.

Reply:
As above, I'm not sure I understand how, since the checks depend on the value of `idx`